docs/developer/release/1-create-release-branch.md

```diff
@@ -23,4 +23,8 @@ Patch Releases for that major pr minor version of the agent.
     - If the consensus commit is not the latest commit from main, branch from that instead.
 
     > **NOTE**: Don't create any other branches that are prefixed with `release` when creating PRs or
-    those branches will collide with our automated release build publish rules.
\ No newline at end of file
+    those branches will collide with our automated release build publish rules.
+
+3. Open a PR against the `main` branch to update the VERSION file to the _next_ release.
+
+    For example, if working on release-v0.31, the VERSION file in main should be updated to v0.32.0.
```

Comment:
I like the idea in general. I'm just not a fan of creating another manual step. PRs like this https://github.com/grafana/agent/pull/5373 may accidentally change this to wrong value since there is 17+ places to update already and it'll be easy to mix up. 

Can we write a short program that will grab one of the versions that we already have in repo, (like these ones https://github.com/grafana/agent/pull/5373) and use semver to increment its minor version? A major release with breaking changes would require some extra manual steps though.

Alternatively, to make it crystal clear what is inside `VERSION`, maybe we should change it to `yaml` or something and have clearly named fields with `latestReleasedVersion` and `nextReleaseVersion`. And then we could instead build some automation where possible to read version from that file and avoid 17+ places to update version in PRs like this https://github.com/grafana/agent/pull/5373.

Reply:
I think it might be a good opportunity for #5333 handle this in addition to all the different places we have to change version.

Reply:
yay! #5333 is awesome

Reply:
I thought about this some last night and there's definitely automation we can introduce to alleviate this extra step. 

I think it should be possible to automatically update this file in main whenever a new release branch is created.